### PR TITLE
Fix the erfcx implementation | fix(torchlib)

### PR DESCRIPTION
**Pull request by justinchuby**
https://github.com/microsoft/onnxscript/issues/1223 but it is still numerically unstable.

### CODE REVIEW

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 1]
```diff
@@ -815,7 +814,9 @@ def _where_input_wrangler(
     TorchLibOpInfo(
         "erfc", special_ops.aten_special_erfc, tolerance={torch.float16: (1e-2, 2e-4)}
     ),
-    # TorchLibOpInfo("erfcx", special_ops.aten_special_erfcx),  # not in OPS_DB
+    TorchLibOpInfo("special.erfcx", special_ops.aten_special_erfcx).xfail(
+        reason="fixme: The implementation is numerical unstable: https://github.com/microsoft/onnxscript/issues/1223"
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:818]:
```suggestion
        reason="fixme: The implementation is numerically unstable: https://github.com/microsoft/onnxscript/issues/1223"
```

File: onnxscript/function_libs/torch_lib/ops/special.py [hunk 2]
```diff
@@ -106,7 +106,7 @@ def aten_special_erfc(self: TFloatOrBFloat16) -> TFloatOrBFloat16:
 def aten_special_erfcx(self: TFloatOrBFloat16) -> TFloatOrBFloat16:
     """special_erfcx(Tensor self) -> Tensor"""
 
-    return op.Mul(op.Exp(op.Pow(self, 2)), op.Erf(self))
+    return op.Mul(op.Exp(op.Pow(self, 2)), op.Sub(1, op.Erf(self)))
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/special.py:109]: Maybe https://math.stackexchange.com/questions/4056083/how-to-address-the-numerical-instability-of-a-function-involving-erf-and-an-expo ?

Comment (titaiwangms) [reply onnxscript/function_libs/torch_lib/ops/special.py:109]: That is what I thought, but it still doesn't match. These are all basic ops and match with pytorch, but when we combine them, it doesn't have enough precision..

**Review by titaiwangms — APPROVED** [review]
Let's merge it with an issue raised in ORT?

File: onnxscript/function_libs/torch_lib/ops/special.py (hunk 2 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/special.py:109]: Do you mean when you use a more numerically stable implementation from this link the results still do not match?

Comment (titaiwangms) [reply onnxscript/function_libs/torch_lib/ops/special.py:111]: Ah, I thought the link was about torch implementation.

**Pull request merged**